### PR TITLE
docs: separate worker and controller led resources

### PR DESCRIPTION
**Pull request by bschaatsbergen**
- **What**: Separated the worker and controlled led `boundary_worker` into two distinct examples.
- **Why**: Combining both the controller and worker LED resources into a single example in the documentation can be confusing.

### CODE REVIEW

File: templates/resources/worker.md.tmpl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+subcategory: ""
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+---
+
+# boundary_worker (Resource)
+
+{{ .Description | trimspace }}
+
+## Controller-led worker
```
Comment (moduli) [inline templates/resources/worker.md.tmpl:12]: Suggestion: What if we organized it to have
```
## Example Usage

### Controller-led worker

### Worker-led worker
```
?

File: examples/resources/boundary_worker/controller_led.tf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+resource "boundary_worker" "controller_led" {
+  scope_id    = "global"
+  name        = "worker 1"
```
Comment (moduli) [inline examples/resources/boundary_worker/controller_led.tf:3]: Should we give the workers different names to reinforce the idea that resources need to have unique names?

Comment (bschaatsbergen) [reply examples/resources/boundary_worker/controller_led.tf:3]: Good catch. I’ll change it to “controller-led-worker-1” and “worker-led-worker-1”.​

File: templates/resources/worker.md.tmpl (hunk 1 above)
Comment (bschaatsbergen) [reply templates/resources/worker.md.tmpl:12]: Fixed 👌🏼

**Review by moduli — APPROVED** [review]

**Pull request merged**